### ISSUE TITLE
WHERE clause in newrelic_infra_alert_condition not reflecting correctly in UI

### ISSUE THREAD

**Issue by skyzyx** [terraform-providers/terraform-provider-newrelic#609]
### Terraform Version

```
Terraform v0.12.26
+ provider.newrelic v1.18.0
```

### Affected Resource(s)

* [newrelic_infra_alert_condition](https://www.terraform.io/docs/providers/newrelic/r/infra_alert_condition.html)

### Terraform Configuration Files

```hcl
# https://www.terraform.io/docs/providers/newrelic/r/alert_policy.html
resource "newrelic_alert_policy" "ops" {
  incident_preference = "PER_CONDITION"
  name                = "testing"
}

# https://www.terraform.io/docs/providers/newrelic/r/infra_alert_condition.html
resource "newrelic_infra_alert_condition" "cond_ec2_instance_cpu" {
  name    = "EC2 Instance CPU"
  enabled = true

  # Bind this condition to the following alert policy
  policy_id = newrelic_alert_policy.ops.id

  # Type of metric and where to read it from
  type       = "infra_metric"
  event      = "SystemSample"
  select     = "cpuPercent"
  comparison = "above"
  where      = "( `ec2Tag_Service` = 'aws-devtools-prod' )"

  # WARN
  warning {
    value         = 80
    duration      = 10
    time_function = "all"
  }

  # CRIT
  critical {
    value         = 90
    duration      = 10
    time_function = "all"
  }

  # Hours until it auto-closes
  violation_close_timer = 1
}
```

### Debug Output

https://gist.github.com/skyzyx/a80928a32f6f7f35ddb79ed182973fe9

### Expected Behavior

When executing this Terraform defining an infrastructure alert condition, I expected the `WHERE` clause to be reflected in the UI (which would confirm my expectation that the alert condition is configured correctly):

<img width="680" alt="Screen Shot 2020-05-28 at 10 50 04 AM" src="https://user-images.githubusercontent.com/39447/83175784-57363200-a0d1-11ea-8849-5b4e6c5fae6d.png">

I also expected the filter to be displayed in the UI for the alert condition itself:

<img width="461" alt="Screen Shot 2020-05-28 at 10 51 46 AM" src="https://user-images.githubusercontent.com/39447/83175855-71701000-a0d1-11ea-8449-dc2afacd4e15.png">

(These screenshots were taken when manually while configuring this alert condition in the UI.)

### Actual Behavior

When executing this Terraform defining an infrastructure alert condition, the `WHERE` clause does not appear to be reflected in the UI, leading me to be uncertain that it even works as intended. No, let me rephrase: _I have no confidence that this alert condition was configured correctly when managed via the Terraform Provider_.

We have an uneditable view in the UI:

<img width="919" alt="Screen Shot 2020-05-28 at 10 46 06 AM" src="https://user-images.githubusercontent.com/39447/83176250-f824ed00-a0d1-11ea-897b-ebd6d660414e.png">

…which is probably fine because we don't want people to be messing with things manually that were configured in Terraform, but it leaves me wondering whether or not this works as expected.

In the UI for the alert condition itself, it doesn't reflect my filter (i.e., `WHERE` clause`), and instead says "All entities" which is definitely **not** what I want.

<img width="432" alt="Screen Shot 2020-05-28 at 10 44 27 AM" src="https://user-images.githubusercontent.com/39447/83176442-4f2ac200-a0d2-11ea-8893-dc281a1ed397.png">

On the other hand, other aspects of my Terraform were applied correctly, and the list of machines in the graph _appear_ to be correct.

<img width="623" alt="Screen Shot 2020-05-28 at 10 46 45 AM" src="https://user-images.githubusercontent.com/39447/83176596-8b5e2280-a0d2-11ea-94cd-ee0e59716c9f.png">

So I'm not sure if this is working, or not working. It doesn't give me any confidence that it does what it's supposed to do.

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:

1. `terraform apply`. The execution should complete without error.

1. Review the alerts and alert conditions to validate the configuration you just pushed.

**Comment by jthurman42** [comment]
@skyzyx Thanks for reporting the issue.  As this is an upstream UI issue, I've created an internal feature request for visibility / parity with what the API reports, and what is displayed in the UI.  Since the provider is unable to perform any action that would resolve this issue, I'm going to close it.  If you require any further information on this item, please open a ticket at https://support.newrelic.com/ Thanks!

**Comment by skyzyx** [comment]
Thanks @jthurman42.

First, I'd like to confirm — _is it_ only a UI issue? Can you confirm that it's not simply an issue with the Terraform Provider or Go SDK? I can't tell from my end if it's (a) the UI not showing the correct thing, or (b) that the data in the system is wrong and the UI is reflecting the wrongness.

Did the `WHERE` get applied or not?

**Comment by skyzyx** [comment]
Confirmed that the API says the condition is in-place.

Here is a snippet of a test using the [Terratest](https://github.com/gruntwork-io/terratest) framework and the [New Relic SDK for Go](https://github.com/newrelic/newrelic-client-go) — where `assert.Equal` resolves to `true`.

```go
// Call out to the SDK
cfg := nrconfig.New()
cfg.AdminAPIKey = os.Getenv("NEWRELIC_API_KEY")
alertsClient := nralerts.New(cfg)

conditionIds := []string{
	"cond_ec2_instance_cpu_id",    // Alert Condition ID from the Terraform output
	"cond_ec2_instance_memory_id", // Alert Condition ID from the Terraform output
}

for i := range conditionIds {
	// https://gowalker.org/github.com/gruntwork-io/terratest/modules/test-structure#LoadString
	conditionID := test_structure.LoadString(t, tmpDir, conditionIds[i])

	// We get back a value that is two numbers, colon-delimited. `00000:00000`. We want the second number.
	conditionID = strings.Split(conditionID, ":")[1]
	condID, err := strconv.Atoi(conditionID)
	if err != nil {
		assert.Fail(t, err.Error())
	}

	// https://gowalker.org/github.com/newrelic/newrelic-client-go/pkg/alerts#Alerts_GetInfrastructureCondition
	condition, err := alertsClient.GetInfrastructureCondition(condID)
	if err != nil {
		assert.Fail(t, err.Error())
	}

	// Make sure that the data we get back from the API is correct
	assert.Equal(t, condition.Where, "( `ec2Tag_Service` = 'aws-devtools-prod' )")
}
```

**Comment by skyzyx** [comment]
@jthurman42: We just got off a call where we were able to talk through this. I'm fine with leaving this closed, and will look into the other options that we discussed around NRQL and performing more thorough testing against higher-traffic apps.

Thanks!

**Comment by bgoleno** [comment]
Hi @skyzyx ,  I am the Product Manager for Alerts, and I am reviewing the feature request that Jon submitted on your behalf.  I want to share some information on our roadmap for this area.  We are in the process of unifying all alert condition types to be "NRQL Conditions".  This includes Infrastructure and APM alerts.  The migration of Infrastructure conditions to NRQL Conditions, and a new streaming alerts infrastructure, has already begun. After the back-end is migrated, we will be making changes to the UI / UX for Alert conditions for Infrastructure Entities. After the back-end migration is complete, which includes moving the Infrastructure data set over to dimensional metrics (aka Open Metrics), you will be able to create these alerts as NRQL conditions directly, and the "WHERE" clause aspect will be explicitly visible in the UI.   
Thanks in advance for your continued patience.

**Comment by skyzyx** [comment]
@bgoleno: Good to know. Thank you for the heads-up. I'll plan on migrating my infrastructure alerts over to NRQL alerts. 👍
